src/victory-legend/victory-legend.js

```diff
@@ -0,0 +1,186 @@
+  color: "#252525",
+  backgroundColor: "#d9d9d9",
+  stroke: "transparent"
+};
+
+const defaultLegendStyle = {
+  height: 200,
+  width: 300,
+  padding: 20
+};
+
+const defaultLegendData = [{
+  name: "Series 1",
+  symbol: {
+    type: "circle",
+    style: {
+      fill: "red"
+    }
+  }
+}, {
+  name: "Series 2",
+  symbol: {
+    type: "diamond",
+    style: {
+      fill: "blue"
+    }
+  }
+}];
+
+export default class VictoryLegend extends React.Component {
+  static propTypes = {
+    x: PropTypes.number.isRequired,
+    y: PropTypes.number.isRequired,
+    height: PropTypes.number.isRequired,
+    width: PropTypes.number.isRequired,
+    orientation: PropTypes.string,
+    data: PropTypes.arrayOf(
+      PropTypes.shape({
+        name: PropTypes.string.isRequired,
+        label: PropTypes.object,
+        symbol: PropTypes.object
+      })
+    ),
+    columnItemSpacing: PropTypes.number,
+    rowItemSpacing: PropTypes.number,
+    groupComponent: PropTypes.element,
+    standalone: PropTypes.bool,
+    style: PropTypes.object
+  };
+
+  static defaultProps = {
+    x: 0,
+    y: 0,
+    height: 100,
+    width: 150,
+    orientation: "vertical",
+    columnItemSpacing: 10,
+    rowItemSpacing: 8,
+    symbolComponent: <Point/>,
+    labelComponent: <VictoryLabel/>,
+    containerComponent: <VictoryContainer/>,
+    groupComponent: <g/>,
+    standalone: true
+  };
+
+  getFontStyles(props) {
+    return merge({}, defaultFontStyle, props.label);
+  }
+
+  getLegendStyle(props) {
+    return {
+      style: merge({}, defaultLegendStyle, props.style)
+    };
+  }
+
+  // it should be changed to approximateTextSize
+  getLabelBbox(text, font) {
+    const body = document.body;
+    const container = document.createElement("svg");
+    const label = document.createElement("text");
+
+    label.innerText = text;
+    label.setAttribute("style",
+      `fill: ${font.color};
+      font-size: ${font.fontSize}px;
+      font-family: ${font.fontFamily};
+      background-color: ${font.backgroundColor};
+      stroke: ${font.stroke};
+      top: -10000px;
+      position: absolute;`
+    );
+    body.appendChild(container);
+    container.appendChild(label);
+
+    const bBox = label.getBoundingClientRect();
+    body.removeChild(container);
+
+    return bBox;
+  }
+
+  renderContainer(props, group) {
+    return React.cloneElement(
+      props.containerComponent,
+      this.getLegendStyle(props),
+      group
+    );
+  }
+
+  renderGroup(children) {
+    return React.cloneElement(
+      this.props.groupComponent,
+      { role: "presentation" },
+      children
+    );
+  }
+
+  render() {
+    let yPadding = 0;
+    let xPadding = 0;
+    const props = this.props;
+    const itemXPosition = [];
+    const symbolComponents = [];
+    const labelComponents = [];
+    const { symbolComponent, labelComponent } = props;
+    const data = isEmpty(props.data) ? defaultLegendData : props.data;
+
+    data.forEach((series, index) => {
```

Comment:
For perf reasons we've been replacing array method that loop across data with length cached for loops